gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -923,6 +923,83 @@ _peer_is_flexible_tensor_caps (GstPad * pad)
   return flexible;
 }
 
+/**
+ * @brief Update caps dimension for negotiation
+ * @param caps caps to compare and update
+ * @param peer_caps caps to compare
+ */
+void
+update_caps_dimension (GstCaps * caps, GstCaps * peer_caps)
+{
+  GstStructure *structure;
+  GstStructure *structure_peer;
+  tensor_dim dim, dim_peer;
+  const gchar *dim_str;
+  const gchar *dim_str_peer;
+  guint i;
+
```

Comment:
How about checking for NULL?

g_return_val_if_fail (caps != NULL, FALSE);
g_return_val_if_fail (peer_caps != NULL, FALSE);

Reply:
updated to check NULL value
Thank you!